ivy_tests/test_ivy/test_frontends/test_torch/test_non_linear_activation_functions.py

```diff
@@ -1078,3 +1079,245 @@ def test_torch_softplus(
+        q_proj_weight = draw(
+            helpers.array_values(
+                dtype=dtype[0],
+                min_value=0.1,
+                max_value=1,
+                shape=(embed_dim, embed_dim),
+            )
+        )
+        k_proj_weight = draw(
+            helpers.array_values(
+                dtype=dtype[0],
+                min_value=0.1,
+                max_value=1,
+                shape=(embed_dim, embed_dim),
+            )
+        )
+        v_proj_weight = draw(
+            helpers.array_values(
+                dtype=dtype[0],
+                min_value=0.1,
+                max_value=1,
+                shape=(embed_dim, embed_dim),
+            )
+        )
+    else:
+        use_separate_proj_weight = False
+        q_proj_weight = None
+        k_proj_weight = None
+        v_proj_weight = None
+    
+    out_proj_weight = draw(
+        helpers.array_values(
+            dtype=dtype[0],
+            min_value=0.1,
+            max_value=1,
+            shape=(embed_dim, embed_dim),
+        )
+    )
+    out_proj_bias = draw(
+        helpers.array_values(
+            dtype=dtype[0],
+            min_value=0.1,
+            max_value=1,
+            shape=(embed_dim,),
+        )
+    )
+    bias_k = random.choice([
+        draw(
+            helpers.array_values(
+                dtype=dtype[0],
+                min_value=0.1,
+                max_value=1,
+                shape=(embed_dim,),
+            )
+        ),
+        None,
+    ])
+    bias_v = bias_k
+
+    if bias_k is None:
+        static_k = random.choice([
+            draw(
+                helpers.array_values(
+                    dtype=dtype[0],
+                    min_value=0.1,
+                    max_value=1,
+                    shape=(batch_size * heads, seq_len, head_dim),
+                )
+            ),
+            None,
+        ])
+        static_v = static_k
+    else:
+        static_k = None
+        static_v = None
+
+    attn_mask = ivy.ones((seq_len, seq_len), dtype=dtype[0])
+    key_padding_mask = random.choice([
+        ivy.random_normal(shape=(seq_len, seq_len), dtype=dtype[0]) > 0,
+        None,
+    ])
+    
+    return (
+        dtype,
+        q,
+        k,
+        v,
+        heads,
+        use_separate_proj_weight,
+        embed_dim,
+        in_proj_weight,
+        in_proj_bias,
+        out_proj_weight,
+        out_proj_bias,
+        q_proj_weight,
+        k_proj_weight,
+        v_proj_weight,
+        bias_k,
+        bias_v,
+        static_k,
+        static_v,
+        attn_mask,
+        key_padding_mask,
+        is_causal,
+    )
+
+
+# multi_head_attention_forward
+@handle_frontend_test(
+    fn_tree="torch.nn.functional.multi_head_attention_forward",
+    dtype_mha_args=mha_forward_args(
+        dtypes=helpers.get_dtypes("float"),
+    ),
+    add_zero_attn=st.just(False),
+    dropout_p=st.just(0.),
+    training=st.just(False),
+    need_weights=st.booleans(),
```

Comment:
Shouldn't this be tested for `training=True` as well? and also for more values of `dropout`?

Reply:
@AnnaTz wouldn't having a dropout other than 0 make the outcome non-deterministic, which would cause the tests to fail?

Reply:
Yeah but I think you should still test for different values and in case the dropout is not zero you can just check that the shape and dtype of the return is correct, like how is done [here](https://github.com/unifyai/ivy/blob/c92849be48c5db478218200667a100404ce43248/ivy_tests/test_ivy/test_functional/test_nn/test_layers.py#L148)

Reply:
@AnnaTz ok that makes sense, I've made a change that should do that. Let me know what you think. Also I had to fix this bug in ivy.dropout to get this to work, so this will have to be merged first if you could take a quick look https://github.com/unifyai/ivy/pull/17516. Thanks 🙂